+sw_tests/+unit_tests/unittest_spinw_optmagsteep.m

```diff
@@ -0,0 +1,223 @@
+classdef unittest_spinw_optmagsteep < sw_tests.unit_tests.unittest_super
+
+    properties
+        swobj = [];
+        default_magstr = struct('S', [0  0; 0 0; -1 1], 'n', [0 0 1], ...
+                                'N_ext', [2 1 1], 'k', [0 0 0], ...
+                                'exact', true)
+        % output from optmagsteep
+        default_opt = struct('M', [0 0; 0 0; -1 1],...
+                             'dM', 6.6e-17,...
+                             'e', -1.1, ...
+                             'nRun', 1, ...
+                             'title', ['Optimised magnetic structure ', ...
+                                      'using the method of steepest ', ...
+                                      'descent']); 
+    end
+    properties (TestParameter)
+        existing_plot = {true, false};
+    end
+    methods (TestMethodSetup)
+        function setup_afm_chain_model_easy_axis(testCase)            
+            testCase.swobj = spinw();
+            testCase.swobj.genlattice('lat_const', [2 3 6])
+            testCase.swobj.addatom('r',[0; 0; 0],'S',1)
+            testCase.swobj.addmatrix('label', 'A', ...
+                                     'value', diag([0 0 -0.1])) % c easy
+            testCase.swobj.addmatrix('label', 'J1', 'value', 1)  % AFM
+            testCase.swobj.gencoupling();
+            testCase.swobj.addcoupling('mat', 'J1', 'bond', 1); % along a
+            testCase.swobj.addaniso('A');
+        end
+    end
+
+    methods (Test)
+        function test_no_nExt_throws_error(testCase)
```

Comment:
It might be more informative to call this `test_only_one_spin_throws_error` - I know in this case it's because `nExt` is not specified, but I think this is because there's only one atom in a unit cell in this example - e.g. I think it would work if you had two `addatom`s in the setup...

Reply:
That's true - I'll also add a test with more than one atom in the unit cell